src/text-input/tests/functional/TextInput.ts

```diff
@@ -120,22 +123,36 @@ registerSuite('TextInput', {
 			.click()
 			.type(validText)
 			.end()
-			.sleep(10)
-			.getProperty('className')
-			.then((className: string) => {
-				assert.notInclude(className, css.invalid);
-				assert.include(className, css.valid);
-			})
-			.findByCssSelector(`.${css.input}`)
-			.type(invalidText)
 			.end()
-			.sleep(10)
-			.getProperty('className')
-			.then((className: string) => {
-				assert.notInclude(className, css.valid);
-				assert.include(className, css.invalid);
+			.then(
+				pollUntilTruthy(
+					function(validClass) {
+						return !!document.querySelector(`#example-validated .${validClass}`);
```

Comment:
Is the goal here to confirm that a previously valid input is no longer valid?

Reply:
Yup that's right, the goal here is to determine when you type "foo" does the widget add a `valid` class, and when you add invalid text "foofoobar" does the widget add the invalid class.